src/auth/googleauth.ts

```diff
@@ -82,11 +84,18 @@ export class GoogleAuth {
 
   cachedCredential: OAuth2Client|null = null;
 
+  private refreshTokenEarlyMillis: number|undefined;
+
   /**
    * Export DefaultTransporter as a static property of the class.
    */
   static DefaultTransporter = DefaultTransporter;
 
+  constructor(options?: GoogleAuthOptions) {
```

Comment:
I'm hesitant to rely on a constructor with this class.  In the near future, I was hoping to look deeply at #4, #54, and #55.  All of these involve the idea of not requiring the user to instantiate this object, but rather providing default exported methods.  Can we experiment with hanging this option of off the relevant methods?

Reply:
I added a setter for `refreshTokenEarlyMillis`

Reply:
I am not very thrilled with this turn of events. I would prefer to keep this to be a construction time option, and not a mutable property.

I would like to understand the needs from the linked issues more deeply before we do a lot more in this direction.

Reply:
@ofrobots and I had a good discussion here.  I think we're both in favor of accepting optional options in the public factory methods on the class.  This has a few benefits:
- We can get rid of the constructor requirement for the `GoogleAuth` class
- We can still make the expiry buffer configurable
- It's still a non-breaking change
- There are no global ramifications 

I think that means extending these methods:
`getApplicationDefault`, `fromJSON`, `fromStream`

Thank you for bearing with us here :)

Reply:
For getApplicationDefaultAsync, should I modify the value for `eagerRefreshThresholdMillis` if cached credentials were used?
Right now I do not.